### PR TITLE
[SWE-26] Profile Page Frontend

### PR DESCRIPTION
**Pull request by datderek**
Task ID: [SWE-26](https://linear.app/swe-dem-boys/issue/SWE-26/profile-page-frontend)
## PR Description
Create a dashboard the displays the user profile picture on the top right. Pressing profile picture routes the user to their profile page. Profile page displays profile picture, follower counter, following count, and post count. Additionally, profile page should conditionally render edit profile button if it belongs to the session user.
## Notes
Uploaded some temporary static files and images in `/data` and `/public/images/pp`. Just added these so I could mimic how the Dynamic Routes and Data Fetching would retrieve routes and static data from the DB.

### CODE REVIEW

File: app/[user]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import Image from 'next/image';
+// These imports just for grabbing the static data - these should be removed
```
Comment (jackfales) [inline app/[user]/page.tsx:2]: In the future use `TODO`. It makes it easier to search the entire codebase for things you need to fix before merging.
```suggestion
// TODO: These imports just for grabbing the static data - these should be removed
```

File: app/[user]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import path from 'path';
+import fs from 'fs';
+
+export const metadata = {
+  title: 'Dashboard'
+}
+
+const profileDirectory: string = path.join(process.cwd(), 'data');
+
+// Grab all users, this would probably query the user database
```
Comment (jackfales) [inline app/[user]/page.tsx:12]: Use [TSDoc](https://tsdoc.org/) syntax for doc comments on functions

File: data/jfales.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "followers": 0,
```
Comment (jackfales) [inline data/jfales.json:2]: Come on man

File: public/images/pp/dtran.jpg
Comment (jackfales) [inline public/images/pp/dtran.jpg:1]: LOL

**Review by jackfales — COMMENTED** [review]
Looks good

File: app/[user]/page.tsx (hunk 1 above)
Comment (datderek) [reply app/[user]/page.tsx:2]: Didn't know about that feature THANKS

File: app/[user]/page.tsx (hunk 2 above)
Comment (datderek) [reply app/[user]/page.tsx:12]: Will do!

File: data/jfales.json (hunk 3 above)
Comment (datderek) [reply data/jfales.json:2]: Please refrain from non-productive comments.

**Review by sammyoto — COMMENTED** [review]
Looks good. @jackfales follower count accurate. Now that I look at it there was no need for a login api page in the first place. Nice catch.

**Review by sammyoto — APPROVED** [review]
Everything looks good to me. We can start migrating to the App router next week. We can talk to Jack tomorrow and make that a task for one of us.

**Review by jackfales — APPROVED** [review]

**Comment by datderek** [comment]
Fixed merge conflicts, just need re-review to merge to main.

**Comment by sammyoto** [comment]
Screen Shot 2023-10-16 at 3 31 30 PM

npm i said critical vulnerability so I ran audit. Should be good now I'll commit the changes.

**Review by sammyoto — APPROVED** [review]
Everything works. Looks good to me.

**Review by jackfales — APPROVED** [review]

**Pull request merged**